test/jdk/java/lang/StableValue/JEP.md

```diff
@@ -0,0 +1,561 @@
+Logger logger = cache.xxxx(7);
+```
+
+and even for several compute-at-most-once values associated with some key of arbitrary type `K` where we use Strings
+as the key type in the example below:
+
+```
+// Declare a cache that can hold a finite number of Logger instance
+// associated with the same finite number of `keys` Strings.
+Baz<String, Logger> cache = ... keys ... Logger::getLogger...
+...
+// Just returns the value associated with the provided string if set. Otherwise computes and returns the value.
+// If another thread is computing a value for the provided String, wait until it has completed.
+// Values can be computed in parallel by distinct threads.
+Logger logger = cache.xxxx("com.company.Foo");
+```
+
+For interoperability with legacy code, it would also be desirable if some of the standard collection types could be
+expressed as compute-at-most-once constructs in a similar fashion:
+
+```
+// Declare a List whose elements are lazily computed once accessed via List::get
+List<Logger> lazyList = ... 10 ... i -> Logger.getLogger("com.company.Foo" + i) ...
+
+// Declare a Map whose values are lazily computed once accessed via Map::get
+Map<String, Logger> lazyMap = ... keys ... Logger::getLogger...
+```
+
+A final note should be made about static fields. Initialization of `static` and `final` fields can be broken up by
+leveraging the laziness already built into class loading. Often referred to as the
+[*initialization-on-demand_holder_idiom*](https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom), this
+technique moves lazily initialized state into a helper class which is then loaded on-demand, so its initialization is
+only performed when the data is actually needed, rather than unconditionally initializing constants when a class is
+first referenced:
+
+```
+// Ordinary static initialization
+private static final Logger LOGGER = Logger.getLogger("com.company.Foo");
+...
+LOGGER.log(Level.DEBUG, ...);
+```
+
+we can defer initialization until we actually need it, like so:
+
+```
+// Initialization-on-demand holder idiom
+Logger logger() {
+    class Holder {
+         static final Logger LOGGER = Logger.getLogger("com.company.Foo");
+    }
+    return Holder.LOGGER;
+}
+...
+logger().log(Level.DEBUG, ...);
+```
+
+The code above ensures that the `Logger` object is created only when actually required. The (possibly expensive)
+initializer for the logger lives in the nested `Holder` class, which will only be initialized when the `logger`
+method accesses the `LOGGER` field. While this idiom works well, its reliance on the class loading process comes
+with significant drawbacks. First, each constant whose computation needs to be deferred generally requires its own
+holder class, thus introducing a significant static footprint cost. Second, this idiom is only really applicable
+if the field initialization is suitably isolated, not relying on any other parts of the object state.
```

Comment:
Maybe also note that this class loading approach is hard to extend at runtime if you add many more constants at runtime.